app/src/main/java/com/duckduckgo/app/global/view/DaxDialog.kt

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2019 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.global.view
+
+import android.app.Dialog
+import android.content.DialogInterface
+import android.graphics.Color
+import android.graphics.Rect
+import android.graphics.drawable.ColorDrawable
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.DialogFragment
+import com.duckduckgo.app.browser.R
+import android.view.Gravity
+import androidx.constraintlayout.widget.ConstraintLayout
+import kotlinx.android.synthetic.main.content_dax_dialog.*
+import android.view.animation.Animation
+import android.view.animation.OvershootInterpolator
+import android.view.animation.ScaleAnimation
+import android.widget.ImageView
+import android.widget.RelativeLayout
+
+class DaxDialog(
+    private val daxText: String,
+    private val buttonText: String,
+    private val dismissible: Boolean = true,
+    private val typingDelayInMs: Long = 20
+) : DialogFragment() {
+
+    private var onAnimationFinished: () -> Unit = {}
+    private var primaryCTAClickListener: () -> Unit = { dismiss() }
```

Comment:
We typically don't keep all uppercase for acronyms.

Suggested change: `primaryCtaClickListener`

Reply:
Done!